### PR TITLE
Expose `Ticker.startTime` so users know the start time and the absolute time

### PR DESCRIPTION
**Pull request by fzyzcjy**
1. I will finish code details, refine code, add tests, make tests pass, etc, after a code review that thinks the *rough idea* is acceptable. It is because, from my past experience, reviews may request changing a lot. If the general idea is to be changed, all detailed implementation efforts are wasted :)
2. The PR has an already-working counterpart, and it produces ~60FPS smooth experimental results. The benchmark results and detailed analysis is in chapter https://cjycode.com/flutter_smooth/benchmark/. All the source code is in  https://github.com/fzyzcjy/engine/tree/flutter-smooth and https://github.com/fzyzcjy/flutter/tree/flutter-smooth.
3. Possibly useful as a context to this PR, there is a whole chapter discussing the internals - how flutter_smooth is implemented. (Link: https://cjycode.com/flutter_smooth/design/)

---

Currently, the user of `Ticker` only knows the `elapsed` time. However, it looks reasonable to allow the user to know when the ticker thinks it *starts* ticking.

If this does not wanted to be widely used, maybe we can mark it as `@protected` or `@visibleForTesting` or [`@experimental`](https://pub.dev/documentation/meta/latest/meta/experimental-constant.html).

As for where it is needed inside flutter_smooth, it is utilized to know the relative time between a few Tickers as well as the system. In other words, when Ticker A says it elapsed 1 second, flutter_smooth needs to know the startTime such that it knows what "1second" means in absolute time. Detailed code can be seen in https://github.com/fzyzcjy/flutter_smooth/blob/0c5db0ff270aa0c8cff28ea19055999627a8df6d/packages/smooth/lib/src/drop_in/list_view/shift.dart#L352-L356
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/scheduler/ticker.dart [hunk 1]
```diff
@@ -137,6 +137,13 @@ class Ticker {
   /// [isTicking].
   bool get isActive => _future != null;
 
+  /// The start time of the ticker.
+  ///
+  /// When [_onTick], the callback is provided with a relative time (the
+  /// "elapsed"). By adding the `elapsed` to this [startTime], you can get the
+  /// absolute time.
+  Duration? get startTime => _startTime;
```
Comment (goderbauer) [inline packages/flutter/lib/src/scheduler/ticker.dart:145]: What does it mean when this returns null?

File: packages/flutter/lib/src/scheduler/ticker.dart [hunk 2]
```diff
@@ -137,6 +137,13 @@ class Ticker {
   /// [isTicking].
   bool get isActive => _future != null;
 
+  /// The start time of the ticker.
```
Comment (goderbauer) [inline packages/flutter/lib/src/scheduler/ticker.dart:140]: The Ticker is designed to measure relative time. I don't see a good reason to change that.

Comment (fzyzcjy) [reply packages/flutter/lib/src/scheduler/ticker.dart:140]: Well it is required for flutter_smooth, mainly here: https://github.com/fzyzcjy/flutter_smooth/blob/0c5db0ff270aa0c8cff28ea19055999627a8df6d/packages/smooth/lib/src/drop_in/list_view/shift.dart#L352-L356

Shortly speaking, I need to know the time that the ListView animation is using, such that I can fully replicate it in an external extra SmoothShift.

**Comment by fzyzcjy** [comment]
So... What should I do now? I have already replied to all questions :)

**Comment by fzyzcjy** [comment]
I see, same as before, I will file soon when having time. Thanks!

**Comment by fzyzcjy** [comment]
The requested issue is created and described the details: https://github.com/flutter/flutter/issues/115901

**Pull request closed without merging**